javapoet/src/main/java/com/palantir/javapoet/TypeSpec.java

```diff
@@ -413,6 +437,18 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
                 firstMember = false;
             }
 
+            // Compact constructor.
+            if (recordConstructor != null
+                    && recordConstructor.code() != null
+                    && !recordConstructor.code().isEmpty()
+                    && recordConstructor.isCompactConstructor()) {
```

Comment:
This is an implementation detail of `MethodSpec` that we shouldn't leak here. This shouldn't be necessary. We should also remove the `MethodSpec.isCompactConstructor` method.

```suggestion
```